Neos.Flow/Classes/ResourceManagement/Streams/PackageStreamWrapper.php

```diff
@@ -0,0 +1,547 @@
+     *       the "@" operator it wouldn't be possible to run file_exists() on a resource without issuing a warning and
+     *       the resulting exception.
+     *
+     * @param string $path The file path or URL to stat. Note that in the case of a URL, it must be a :// delimited URL. Other URL forms are not supported.
+     * @param integer $flags Holds additional flags set by the streams API.
+     * @return array Should return as many elements as stat() does. Unknown or unavailable values should be set to a rational value (usually 0).
+     */
+    public function pathStat($path, $flags)
+    {
+        $evaluatedPath = $this->evaluatePackagePath($path);
+        if ($evaluatedPath === false) {
+            return false;
+        }
+        return @stat($evaluatedPath);
+    }
+
+    /**
+     * Evaluates the absolute path and filename of the package file specified
+     * by the given path.
+     *
+     * @param string $requestedPath
+     * @param boolean $checkForExistence Whether a (non-hash) path should be checked for existence before being returned
+     * @return mixed The full path and filename or false if the file doesn't exist
+     * @throws \InvalidArgumentException|ResourceException
+     */
+    protected function evaluatePackagePath($requestedPath, $checkForExistence = true)
+    {
+        $requestPathParts = explode('://', $requestedPath, 2);
+        if ($requestPathParts[0] !== self::SCHEME) {
+            throw new \InvalidArgumentException('The ' . __CLASS__ . ' only supports the \'' . self::SCHEME . '\' scheme.', 1256052544);
+        }
+
+        if (!isset($requestPathParts[1])) {
+            return false;
+        }
+
+        $resourceUriWithoutScheme = $requestPathParts[1];
+
+        list($packageName, $path) = explode('/', $resourceUriWithoutScheme, 2);
+
+        try {
+            $package = $this->packageManager->getPackage($packageName);
+        } catch (\Neos\Flow\Package\Exception\UnknownPackageException $packageException) {
+            throw new ResourceException(sprintf('Invalid resource URI "%s": Package "%s" is not available.', $requestedPath, $packageName), 1309269952, $packageException);
+        }
+
+        if (!$package instanceof FlowPackageInterface) {
+            return false;
+        }
+
+        if (!$this->pathRespectsTraversalBoundaries($path)) {
+            throw new \InvalidArgumentException('The ' . __CLASS__ . ' in the \'' . self::SCHEME . '\' scheme. Does not allow to traverse out of the package.', 1665317583);
+        }
+
+        $resourceUri = Files::concatenatePaths([$package->getPackagePath(), $path]);
+
+        if ($checkForExistence === false || file_exists($resourceUri)) {
+            return $resourceUri;
+        }
+
+        return false;
+    }
+
+    private function pathRespectsTraversalBoundaries(string $path): bool
+    {
+        $path = Files::getUnixStylePath($path);
```

Comment:
```suggestion
        $path = Files::getUnixStylePath($path);
        if (str_contains($path, '/../') === false) {
            return true;
        }
```

Shortcut -> as using `..` is more like the exception.

Reply:
Would a path ending with `/..` be rejected elsewhere?